src/main/java/com/xantar/xantarcore/db/MealsService.java

```diff
@@ -40,6 +42,37 @@ public Meal createMeal(Meal meal) {
 		return EMealMapper.toModel(this.mealsRepository.save(eMeal));
 	}
 
+	public Meal updateMeal(Meal updatedData) {
+		if(updatedData == null || updatedData.id == null) {
+			throw new IllegalArgumentException("Cannot update null value as meal");
+		}
+
+		final EMeal originalEMeal = this.mealsRepository.findById(updatedData.id)
+				.orElseThrow(() -> new EntityNotFoundException("Meal with id ''" + updatedData.id));
+
+		final EMeal updateEMeal = this.updateEntity(originalEMeal, updatedData);
+
+		return EMealMapper.toModel(this.mealsRepository.save(updateEMeal));
+	}
+
+	/**
+	 * Updates just non null values
+	 * **/
+	private EMeal updateEntity(EMeal originalEMeal, Meal updatedData) {
+		final EMeal eMeal =  new EMeal.EMealBuilder()
+				.withId(originalEMeal.id)
+				.withName(updatedData.name != null ? updatedData.name : originalEMeal.name)
```

Comment:
What if the name is empty? ('') 
Should this be allowed?

Reply:
As discussed, no, it should not be allowed.